doc/grammar.txt

```diff
@@ -60,25 +61,25 @@ primarySuffix = '(' (exprColonEqExpr comma?)* ')'
       | DOTLIKEOP optInd symbol generalizedLit?
       | '[' optInd exprColonEqExprList optPar ']'
       | '{' optInd exprColonEqExprList optPar '}'
-      | &( '`'|IDENT|literal|'cast'|'addr'|'type') expr (comma expr)* # command syntax
 pragma = '{.' optInd (exprColonEqExpr comma?)* optPar ('.}' | '}')
 identVis = symbol OPR?  # postfix position
 identVisDot = symbol '.' optInd symbol OPR?
 identWithPragma = identVis pragma?
 identWithPragmaDot = identVisDot pragma?
 declColonEquals = identWithPragma (comma identWithPragma)* comma?
-                  (':' optInd typeDesc)? ('=' optInd expr)?
+                  (':' optInd typeDescExpr)? ('=' optInd expr)?
 identColonEquals = IDENT (comma IDENT)* comma?
-     (':' optInd typeDesc)? ('=' optInd expr)?)
-tupleDecl = 'tuple'
-    '[' optInd  (identColonEquals (comma/semicolon)?)*  optPar ']' |
-    COMMENT? (IND{>} identColonEquals (IND{=} identColonEquals)*)?
+     (':' optInd typeDescExpr)? ('=' optInd expr)?)
+tupleType = 'tuple' '[' optInd (identColonEquals (comma/semicolon)?)* optPar ']'
+tupleDecl = tupleType |
+    'tuple' COMMENT? (IND{>} identColonEquals (IND{=} identColonEquals)*)?
```

Comment:
Overlapping first set: `tuple` cannot be disambiguated with an LL(1) infrastucture hence these rules cannot reflect reality.

In other words, it needs to be refactored to:

```suggestion
tupleTypeBracket = '[' optInd (identColonEquals (comma/semicolon)?)* optPar ']'
tupleDecl = 'tuple' (tupleTypeBracket |
    COMMENT? (IND{>} identColonEquals (IND{=} identColonEquals)*)?)
```